lib/agent0/agent0/hyperdrive/policies/lpandarb.py

```diff
@@ -0,0 +1,642 @@
+                interface=interface,
+            )
+            bonds_needed = -bonds_needed  # we trade positive numbers around here
+            # Start by reducing shorts
+            if len(wallet.shorts) > 0:
+                for maturity_time, short in wallet.shorts.items():
+                    reduce_short_amount = min(short.balance, bonds_needed)
+                    bonds_needed -= reduce_short_amount
+                    print(f"reducing short by {reduce_short_amount}")
+                    action_list.append(
+                        Trade(
+                            market_type=MarketType.HYPERDRIVE,
+                            market_action=HyperdriveMarketAction(
+                                action_type=HyperdriveActionType.CLOSE_SHORT,
+                                trade_amount=reduce_short_amount,
+                                wallet=wallet,
+                                maturity_time=maturity_time,
+                            ),
+                        )
+                    )
+            # Open a new long, if there's still a need, and we have money
+            if we_have_money and bonds_needed > MIN_TRADE_AMOUNT:
+                max_long_bonds = interface.get_max_long(wallet.balance.amount)
+                max_long_shares, _, _ = get_shares_in_for_bonds_out(
+                    interface.pool_info["bondReserves"],
+                    interface.pool_info["sharePrice"],
+                    interface.pool_config["initialSharePrice"],
+                    interface.pool_info["shareReserves"],
+                    max_long_bonds,
+                    interface.pool_config["timeStretch"],
+                    interface.pool_config["curveFee"],
+                    interface.pool_config["governanceFee"],
+                )
+                amount = min(shares_needed, max_long_shares) * interface.pool_info["sharePrice"]
+                action_list.append(
+                    Trade(
+                        market_type=MarketType.HYPERDRIVE,
+                        market_action=HyperdriveMarketAction(
+                            action_type=HyperdriveActionType.OPEN_LONG,
+                            trade_amount=amount,
+                            wallet=wallet,
+                        ),
+                    )
+                )
+
+        # Low fixed rate detected
+        if low_fixed_rate_detected:
+            shares_needed, bonds_needed = calc_reserves_to_hit_target_rate(
+                target_rate=interface.variable_rate,
+                interface=interface,
+            )
+            print(f"{bonds_needed=}")
+            # Start by reducing longs
+            if len(wallet.longs) > 0:
+                for maturity_time, long in wallet.longs.items():
+                    reduce_long_amount = min(long.balance, bonds_needed)
+                    bonds_needed -= reduce_long_amount
+                    print(f"reducing long by {reduce_long_amount}")
```

Comment:
log

Reply:
all of these prints should probably be `logging.debug` -- this is too much for the `info` channel.